src/test/java/github/grace000/erversay/MockConnection.java

```diff
@@ -0,0 +1,40 @@
+package github.grace000.erversay;
+
+public class MockConnection implements ConnectionWrapper {
+    private boolean listenForConnectionCalled = false;
+    private boolean acceptConnectionCalled = false;
+    private boolean getIOStreamsCalled = false;
+    private boolean writeResponseCalled = false;
+
+    public void listenForConnection(int port) {
+        listenForConnectionCalled = true;
+    }
+
+    public void serveClient(String response) {
+        acceptConnectionCalled = true;
+        getIOStreamsCalled = true;
+        writeResponseCalled = true;
+    }
+
+    public void close() {
+
+    }
```

Comment:
What's this here for? Doesn't look like it's used.

Reply:
I deleted that, but the tests would not run without the mock implementing all of the methods in the connection wrapper.